internal/controller/git/git.go

```diff
@@ -48,19 +48,27 @@ type Repo interface {
 	Checkout(branch string) error
 	// Commit commits staged changes to the current branch.
 	Commit(message string) error
+	// CommitsHaveDiffs returns whether there is a diff between two commits/branches
+	CommitsHaveDiffs(commit1 string, commit2 string) (bool, error)
```

Comment:
Nit: Since this is described as also being effective for comparing differences between branches, it's possible that the name `RefsHaveDiffs` might be more inclusive of the ways this could be used.